integration/integration_test.go

```diff
@@ -4039,8 +4039,21 @@ func testDiscovery(t *testing.T, suite *integrationTestSuite) {
 	helpers.WaitForActiveTunnelConnections(t, main.Tunnel, "cluster-remote", 1)
 	helpers.WaitForActiveTunnelConnections(t, secondProxy, "cluster-remote", 1)
 
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		// once the tunnel is established we need to wait until we have a
+		// connection to the remote auth
+		site := main.GetSiteAPI("cluster-remote")
+		if !assert.NotNil(c, site) {
+			return
+		}
+		// we need to wait until we know about the node because direct dial to
+		// unregistered servers is no longer supported
+		_, err := site.GetNode(ctx, apidefaults.Namespace, main.Config.HostUUID)
+		assert.NoError(c, err)
+	}, time.Minute, 250*time.Millisecond)
```

Comment:
`runCommand` has a retry count; we could've just made it more generous, but I felt like it was better to split the responsibility for the retries on the different moving parts here.

Reply:
It doesn't seem to be a problem so far, but if the flakiness becomes more widespread we can fold a separate check for the existence of the node in `runCommand` instead, so that at least we get some better error messages.